tests/unit/conftest.py

```diff
@@ -135,14 +136,40 @@ def restart_savepoint(session, transaction):  # pylint:disable=unused-variable
         ):  # pylint:disable=protected-access
             session.begin_nested()
 
+    set_factory_boy_sqlalchemy_session(session)
+
     try:
         yield session
     finally:
+        clear_factory_boy_sqlalchemy_session()
         session.close()
         trans.rollback()
         conn.close()
 
 
+def sqlalchemy_factory_classes():
+    # Return all the SQLAlchemy factory classes from tests.factories.
+    for factory_class in factories.__dict__.values():
+        try:
+            is_sqla_factory = issubclass(factory_class, SQLAlchemyModelFactory)
+        except TypeError:
+            is_sqla_factory = False
+
+        if is_sqla_factory:
+            yield factory_class
```

Comment:
I _think_ this can be replaced with:

```suggestion
    return SQLAlchemyModelFactory.__subclasses__()
```

If so, it might be worth in-lining below?

Reply:
We don't want _all_ subclasses of `SQLAlchemyModelFactory` though, we just want our own subclasses. If for example Factory Boy subclasses that itself internally for some reason we wouldn't want to mess with that.

This would work if we had our own `SQLAlchemyModelFactory` subclass as a base class for all our SQLAlchemy models but then we'd need to implement that base class and remember to always use it instead of the default

Reply:
I'm pretty sure that it doesn't at the moment, but fair enough. Likely a list comprehension that would filter those out too if you're worried.